lib/trip.rb

```diff
@@ -1,21 +1,43 @@
 require 'csv'
+require 'time'
 
 module RideShare
   class Trip
     attr_reader :id, :passenger, :driver, :start_time, :end_time, :cost, :rating
 
     def initialize(input)
+
       @id = input[:id]
       @driver = input[:driver]
       @passenger = input[:passenger]
-      @start_time = input[:start_time]
-      @end_time = input[:end_time]
-      @cost = input[:cost]
-      @rating = input[:rating]
+      @start_time = input[:start_time] == nil ? Time.now : input[:start_time]
+      @end_time = input[:end_time] == nil ? nil : input[:end_time]
+      @cost = input[:cost] == nil ? nil : input[:cost]
+      @rating = input[:rating] == nil ? nil : input[:rating]
```

Comment:
All these checks work fine, but I believe the recommended style in ruby is to use the `.nil?` method:

```ruby
@start_time = input[:start_time].nil? Time.now : input[:start_time]
```

Also, you could omit the check for `end_time`, `cost` and `rating`, since you would keep the `nil` value either way.